### PR TITLE
Share photo

### PR DESCRIPTION
**Pull request by johnmarksinclair**
## Purpose

TPY-22 Share photo to WhatsApp, Reddit, Instagram, etc.

## Proposed Change

Implement share functionality for final stylised image.

## Checklist

- [x] implemented share functionality
- [x] modified routing - split up take/ upload photo page and choose style page
- [x] various styling changes

### CODE REVIEW

File: screens/ChooseStyleScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { StyleSheet, ImageBackground, Image, ScrollView, Dimensions } from 'react-native'
+import { Text, View } from 'components/Themed'
+import Button from 'components/Button'
+import ThemePicker from 'components/ThemePicker'
+import ProjectThemes, { IProjectTheme } from 'constants/ProjectThemes'
+
+const { width, height } = Dimensions.get('screen')
+const containerWidth = width * 0.8
+
+export default function ChooseStyleScreen({ route, navigation }) {
```
Comment (DanielNugent) [inline screens/ChooseStyleScreen.tsx]: could add types for the function parameters so it knows what to expect

**Review by DanielNugent — DISMISSED** [review]

**Review by dabreadman — COMMENTED** [review]
I "worked" on it so it must be fine!

That being said there are a few things to add (as an issue):
- You can simply refuse to log in and still use that application as per usual
- Capture image buttons are off-screen in some Android devices (mine!)

Won't approve now since once you push the change requested by Daniel, you'd dismiss the approval
image

File: screens/ChooseStyleScreen.tsx (hunk 1 above)
Comment (johnmarksinclair) [reply screens/ChooseStyleScreen.tsx]: done

File: screens/FinishedArtScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { StyleSheet, Image, Dimensions } from 'react-native'
+import { Text, View } from 'components/Themed'
+import Button from 'components/Button'
+import * as Sharing from 'expo-sharing'
+
+const { width, height } = Dimensions.get('screen')
+const containerWidth = width * 0.8
+
+interface Props {
```
Comment (DanielNugent) [inline screens/FinishedArtScreen.tsx]: Could put in the actual types

```
import type { NativeStackScreenProps } from '@react-navigation/native-stack'

type RootStackParamList = {
  FinishedArtScreen: { image: string }
  ....
}

type Props = NativeStackScreenProps<RootStackParamList, 'FinishedArtScreen'>
```

https://reactnavigation.org/docs/typescript/

**Review by DanielNugent — APPROVED** [review]

**Pull request merged**